README.md

```diff
@@ -81,6 +91,19 @@ The `Stack` class is a **linked list-based** implementation of a stack, used to
 
 - **`pop(self)`**: Removes and returns the grid and score from the top of the stack.
 
+### TreePossibilities Class
+- Generates a tree of possible moves.
+- Analyzes potential outcomes for optimal move suggestions.
+
+### Ranking System 
+- Tracks high scores.
+- Sorts scores using QuickSort and Bubble Sort.
+- Top scores are stored and displayed to the user.
+
+### Tree-Based Move Suggestion
+- Provides the best move suggestions based on tree analysis.
```

Comment:
(Provides the best move suggestions based on tree analysis.) it is based on higher score, the higher score is detected => so we advice that move